agent/src/beerocks/monitor/monitor_stats.cpp

```diff
@@ -181,7 +181,12 @@ void monitor_stats::process()
             sta_stats_msg.rx_rssi           = sta_stats.rx_rssi_curr;
         }
 
-        beerocks_header->actionhdr()->id() = requests_list.front();
+        auto actionhdr = beerocks_header->actionhdr();
+        if (!actionhdr) {
+            return;
+        }
+
+        actionhdr->id() = requests_list.front();
```

Comment:
The rest of the commit is an almost mechanical change `foo == nullptr` -> `!foo`. Therefore it would be better to do this bit in a separate commit.